RecipeMatch/View Controllers/ProfileViewController.m

```diff
@@ -87,38 +88,18 @@ - (NSInteger)collectionView:(nonnull UICollectionView *)collectionView numberOfI
     return self.recipes.count;
 }
 
-- (void)fetchRecipeImageWithUrl:(NSURL*)url andCompletion:(void (^_Nullable)(BOOL succeeded, UIImage *data))completion {
-    NSURLSessionTask *task = [[NSURLSession sharedSession] dataTaskWithURL:url completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
-        if (data) {
-            // load image data asynchronously
-            UIImage *image = [UIImage imageWithData:data];
-            if (image) {
-                dispatch_async(dispatch_get_main_queue(), ^{
-                    completion(YES, image);
-                });
-            }
-        }
-    }];
-    [task resume];
-    completion(NO, nil);
+// called by cellForItemAtIndexPath to set up cell with image and title
+- (GridRecipeCell *)setupCell:(GridRecipeCell *)cell withRecipe:(SavedRecipe *)recipe {
+    [cell.imageView sd_setImageWithURL:[NSURL URLWithString:recipe[@"image"]] placeholderImage:[UIImage systemImageNamed:@"photo"]];
+    cell.imageView.layer.cornerRadius = CORNER_RADIUS;
+    cell.recipeTitle.text = recipe[@"name"];
+    return cell;
```

Comment:
few things.
- `recipe` is already a model object, so you can access it's image\name with `recipe.name`
-  `cell` is being passed to here (and not a copy), so no need to return it back
-  it's better to move this setup functionality to the cell it self, and do all of this from the there, and in `cellForItemAtIndexPath ` just call `[cell setupWithRecipe:self.recipes[indexPath.row]]`